// ngx-disqus/esm2015/lib/disqus.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import { DISQUS_SHORTNAME } from './disqus.model';
import * as i0 from "@angular/core";
import * as i1 from "./disqus.model";
import * as i2 from "@angular/common";
export class DisqusService {
    /**
     * @param {?} shortname
     * @param {?} _document
     */
    constructor(shortname, _document) {
        this.shortname = shortname;
        this._document = _document;
    }
    /**
     * @return {?}
     */
    get DISQUS() {
        return this._document.defaultView.DISQUS;
    }
    /**
     * @return {?}
     */
    get disqus_config() {
        return this._document.defaultView.disqus_config;
    }
    /**
     * @param {?} config
     * @return {?}
     */
    set disqus_config(config) {
        this._document.defaultView.disqus_config = config;
    }
}
DisqusService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
DisqusService.ctorParameters = () => [
    { type: String, decorators: [{ type: Inject, args: [DISQUS_SHORTNAME,] }] },
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
/** @nocollapse */ DisqusService.ngInjectableDef = i0.defineInjectable({ factory: function DisqusService_Factory() { return new DisqusService(i0.inject(i1.DISQUS_SHORTNAME), i0.inject(i2.DOCUMENT)); }, token: DisqusService, providedIn: "root" });
if (false) {
    /** @type {?} */
    DisqusService.prototype.shortname;
    /** @type {?} */
    DisqusService.prototype._document;
}
