// angular6-social-button/esm5/lib/entities/baseLoginProviderImpl.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
/**
 * @abstract
 */
var /**
 * @abstract
 */
BaseLoginProviderImpl = /** @class */ (function () {
    function BaseLoginProviderImpl() {
    }
    /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    BaseLoginProviderImpl.prototype.loadScript = /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    function (obj, onload) {
        if (document.getElementById(obj.name)) {
            return;
        }
        /** @type {?} */
        var signInJS = document.createElement('script');
        signInJS.async = true;
        signInJS.src = obj.url;
        signInJS.onload = onload;
        if (obj.name === 'linkedin') {
            signInJS.async = false;
            signInJS.text = ('api_key: ' + obj.id).replace('\'', '');
        }
        document.head.appendChild(signInJS);
    };
    return BaseLoginProviderImpl;
}());
/**
 * @abstract
 */
export { BaseLoginProviderImpl };
if (false) {
    /** @type {?} */
    BaseLoginProviderImpl.prototype.TYPE;
    /**
     * @abstract
     * @return {?}
     */
    BaseLoginProviderImpl.prototype.initialize = function () { };
    /**
     * @abstract
     * @param {?=} scopes
     * @return {?}
     */
    BaseLoginProviderImpl.prototype.signIn = function (scopes) { };
    /**
     * @abstract
     * @return {?}
     */
    BaseLoginProviderImpl.prototype.signOut = function () { };
    /**
     * @abstract
     * @param {?=} share
     * @return {?}
     */
    BaseLoginProviderImpl.prototype.sharing = function (share) { };
}
